avocado/utils/wait.py

```diff
@@ -4,7 +4,7 @@
 log = logging.getLogger('avocado.test')
 
 
-def wait_for(func, timeout, first=0.0, step=1.0, text=None):
+def wait_for(func, timeout, first=0.0, step=1.0, text=None, *args, **kwargs):
```

Comment:
The problem with that syntax is that including non-keyword arguments after keyword arguments will break the execution.

If I already have:
```python
def is_ready():
    return True
wait.wait_for(is_ready, timeout=10)
```
Including the function argument:
```python
def is_ready(status):
    return status
wait.wait_for(is_true, timeout=10, True)
```
Will result in `SyntaxError: non-keyword arg after keyword arg`. So, I'd have to rewrite the call, including all the keyword arguments, to then include the function argument, like this:

```python
def is_ready(status):
    return status
wait.wait_for(is_true, 10, 0.0, 1.0, None, True)
```
Not very practical.

Instead, we should allow the args and kwargs to be provided as a list and a dictionary, with proper keywords:

```python
def wait_for(func, timeout, first=0.0, step=1.0, text=None, args=[], kwargs={}):
    ...
        output = func(*args, **kwargs) 
```
So the call would be:
```python
def is_ready(status):
    return status
wait.wait_for(is_true, timeout=10, args=[True])
```
or...
```python
def is_ready(status=False):
    return status
wait.wait_for(is_true, timeout=10, kwargs={'status': True})
```

Reply:
Thanks for your review.  this is our usage:
```
>>> import wait
>>> def is_ready(status):
...     return status
... 
>>> wait.wait_for(is_ready, timeout=10, status=True)
True
```

Is it regular?

CC @lmr

Reply:
Yes, it works because you're using a keyword argument. Try with a non-keyword one (`*args`).

Reply:
Well, actually it will not even work as you described...
```
>>> def is_ready(status):
...     return status
... 
>>> wait.wait_for(is_ready, timeout=10, status=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: wait_for() got an unexpected keyword argument 'status'
```

Reply:
@apahim are you applied this PR?

```
[amos@amos-centos7 utils]$ grep wait_for wait.py
def wait_for(func, timeout, first=0.0, step=1.0, text=None, *args, **kwargs):
[amos@amos-centos7 utils]$ python
Python 2.7.5 (default, Aug  4 2017, 00:39:18) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import wait
>>> def is_ready(status):
...     return status
... 
>>> wait.wait_for(is_ready, timeout=10, status=True)
True
```

Reply:
@apahim I agree on you `args=[], kwargs={}` is better, it's very clear.
avocado/utils/data_structures.py: register() / unregister() has similar usage.

I will update the PR now, thanks.

Reply:
Oh, indeed. It works, but anyways... the issue with non-keyword args is still there and my suggestion stays the same.

Reply:
Ok, thanks!